### ISSUE TITLE
depends_on with health checks (requires compose.yml v3)

### ISSUE THREAD

**Issue by aduzsardi** [faradayio/cage#106]
Hi,
Can you have dependencies between services in a pod ?
like serviceA should start before serviceB where both services are in the same yaml file

**Comment by emk** [comment]
Yes, within a pod you can use [`depends_on`](https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on) normally.

**Comment by aduzsardi** [comment]
Thank you , i forgot i can use only v2 compose directives  so no heath check depends_on
```
    depends_on:
      api:
        condition: service_healthy
```

**Comment by emk** [comment]
Ah, yes, that would be a nice thing to support. Is it v3 only?

**Comment by aduzsardi** [comment]
Yes , it's v3 only

**Comment by emk** [comment]
Right now, we support multiple versions of v2, but we don't really have any APIs for supporting v3, and I'm not even sure how I should handle v2 and v3 support. Migrate everything to v3 internally when we load it, maybe? Hmm.

**Comment by aduzsardi** [comment]
I don't know why i was under the impression that it's only in v3 , seems like i was mistaken
https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on
